consensus/polybft/consensus_runtime.go

```diff
@@ -996,6 +998,9 @@ func (c *consensusRuntime) IsValidBlock(proposal []byte) bool {
 }
 
 func (c *consensusRuntime) IsValidSender(msg *proto.Message) bool {
+	c.lock.Lock()
```

Comment:
is it Lock or Rlock?

Reply:
good catch, I think it can be RL on all checks.

Reply:
agree, only `c.fsm = fsm` at the very beginning of the PR should be a write lock.